File: chatbot/services/brevo.py
"""Brevo transactional email — bare httpx, no SDK."""
import httpx
from config import config


def send_email(subject: str, html_body: str, to: str | None = None) -> bool:
    if not config.BREVO_API_KEY:
        print("[brevo] missing API key, skipping send")
        return False
    target = to or config.BREVO_TO_EMAIL
    try:
        r = httpx.post(
            "https://api.brevo.com/v3/smtp/email",
            headers={
                "accept": "application/json",
                "api-key": config.BREVO_API_KEY,
                "content-type": "application/json",
            },
            json={
                "sender": {"name": "Magistrum Chatbot", "email": config.BREVO_FROM_EMAIL},
                "to": [{"email": target}],
                "subject": subject,
                "htmlContent": html_body,
            },
            timeout=10.0,
        )
        return r.status_code < 300
    except Exception as e:
        print(f"[brevo] send failed: {e}")
        return False

File: web/lib/brevo.ts
// Lightweight Brevo (Sendinblue) transactional email client.
// Uses fetch, no SDK dependency.

type SendArgs = {
  subject: string;
  htmlBody: string;
  toEmail?: string;
};

export async function sendBrevoEmail({ subject, htmlBody, toEmail }: SendArgs): Promise<boolean> {
  const apiKey = process.env.BREVO_API_KEY;
  const fromEmail = process.env.BREVO_FROM_EMAIL;
  const defaultTo = process.env.BREVO_TO_EMAIL;
  const to = toEmail || defaultTo;
  if (!apiKey || !fromEmail || !to) {
    console.warn("[brevo] Missing env config; skipping email send.");
    return false;
  }
  try {
    const r = await fetch("https://api.brevo.com/v3/smtp/email", {
      method: "POST",
      headers: {
        accept: "application/json",
        "api-key": apiKey,
        "content-type": "application/json",
      },
      body: JSON.stringify({
        sender: { name: "Magistrum Pitch Site", email: fromEmail },
        to: [{ email: to }],
        subject,
        htmlContent: htmlBody,
      }),
    });
    if (!r.ok) {
      console.error("[brevo] failed", r.status, await r.text());
      return false;
    }
    return true;
  } catch (e) {
    console.error("[brevo] exception", e);
    return false;
  }
}
